runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/RegisterNodeFunction.java

```diff
@@ -340,6 +335,53 @@ public Node apply(MutableNetwork<Node, Edge> input) {
             }
             ptransformIdToPCollectionViews.put(ptransformId, pcollectionViews.build());
 
+            // This gets the main input pcollection id for this PTransform. This will use the id to
```

Comment:
Please, extract this part to a separate method with descriptive name (createAndInjectTimerCoder?).

Splitting contents of that method is appreciated as well.